library/ssl_ticket.c

```diff
@@ -114,6 +114,7 @@ static int ssl_ticket_gen_key( mbedtls_ssl_ticket_context *ctx,
 /*
  * Rotate/generate keys if necessary
  */
+#if defined(MBEDTLS_CIPHER_MODE_AEAD) || defined(MBEDTLS_NIST_KW_C)
```

Comment:
I understand this makes sense as this is what guards `mbedtls_cipher_auth_encrypt_ext()` but in this context only AEAD is allowed - `mbedtls_ssl_ticket_setup()` enforces that.

Reply:
After optimizing the configurations I think this can be reverted.